hosts/build02/nixpkgs-update.nix

```diff
@@ -86,21 +115,26 @@ let
       User = "r-ryantm";
       Group = "r-ryantm";
       Restart = "on-failure";
-      RestartSec = "5s";
-      WorkingDirectory = "/var/lib/nixpkgs-update/";
+      RestartSec = "30m";
+      LogsDirectory = "nixpkgs-update/";
+      LogsDirectoryMode = "755";
       StateDirectory = "nixpkgs-update";
       StateDirectoryMode = "700";
       CacheDirectory = "nixpkgs-update/worker";
       CacheDirectoryMode = "700";
     };
-    script = ''
-      pipe=/var/lib/nixpkgs-update/fifo
-
-      if [[ ! -p $pipe ]]; then
-        mkfifo $pipe || true
-      fi
 
-      ${cmd} | sort -R > $pipe
```

Comment:
Could we restore randomisation? Two scenarios I think it would help with:

Duplicate PRs (same version bump, different attribute name) opened by separate workers at the same time.

Sometimes going multiple hours without opening a PR when it working through package sets. I'm reluctant to expand the bot skiplist or start putting `DO NOT UPDATE` on stuff in nixpkgs as sometimes it just seems to be transient breakage and also having recent failure logs could be helpful in fixing actual problems.

Reply:
I think we could do something that fuzzes the initial order in which newly ingested packages are queued up. After the first full go-around, existing packages will be dispatched in roughly the same order every time, but for the things you're pointing out I don't think that should matter, yeah?

Reply:
Yes, that should be fine. Thanks for implementing it.

Reply:
Not sure what happened here, submitted four PRs within a few minutes for the same vaultwarden update.

![vaultwarden](https://github.com/nix-community/infra/assets/59103226/7669f4e5-4355-44fd-ae00-4e6283fd027e)

Reply:
It's possible there's a bug, but it's also possible for this to happen organically as designed. If a fetcher emits any never-before-seen attribute paths, the supervisor still prioritizes those over attribute paths it has attempted to update before. If there are only a small number of such paths for a given run of a fetcher, they're going to get bunched together.